async_producer.go

```diff
@@ -200,16 +200,9 @@ func (p *asyncProducer) AsyncClose() {
 	go withRecover(p.shutdown)
 }
 
-///////////////////////////////////////////
-// In normal processing, a message flows through the following functions from top to bottom,
-// starting at topicDispatcher (which reads from Producer.input) and ending in flusher
-// (which sends the message to the broker). In cases where a message must be retried, it goes
-// through retryHandler before being returned to the top of the flow.
-///////////////////////////////////////////
-
```

Comment:
Ya. Once this is merged I'll try and find time for "part 2" which will be a similar process for the aggregator and flusher. At that point this comment can come back, excepti "flows through the following structs, via each `dispatch()` method".